src/librustc_session/options.rs

```diff
@@ -480,18 +481,16 @@ macro_rules! options {
         }
 
         fn parse_sanitizer_memory_track_origins(slot: &mut usize, v: Option<&str>) -> bool {
-            match v.map(|s| s.parse()) {
-                None => {
-                    *slot = 2;
-                    true
-                }
-                Some(Ok(i)) if i <= 2 => {
-                    *slot = i;
-                    true
-                }
-                _ => {
-                    false
-                }
+            match v {
+                Some(s) => {
+                    match s {
+                        "0" => { *slot = 0; true }
+                        "1" => { *slot = 1; true }
+                        "2" => { *slot = 2; true }
+                        _ => { false }
+                    }
+                },
```

Comment:
```suggestion
                Some("0") => { *slot = 0; true }
                Some("1") => { *slot = 1; true }
                Some("2") => { *slot = 2; true }
                Some(_) => false,
```

Reply:
Nice suggestion. And I can combine the `Some("2")` and `None` arms.